src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -370,7 +371,7 @@ internal bool VerifySetDefaults(CookieVariant variant, Uri uri, bool isLocalDoma
             }
 
             // Check the name
-            if (m_name == null || m_name.Length == 0 || m_name[0] == '$' || m_name.IndexOfAny(ReservedToName) != -1)
+            if (string.IsNullOrEmpty(m_name) || m_name[0] == '$' || m_name.IndexOfAny(ReservedToName) != -1 || m_name.Trim() != m_name)
```

Comment:
Could allocate on trim
```suggestion
            if (m_name == null || m_name.Length == 0 || m_name[0] == '$' || m_name.IndexOfAny(ReservedToName) != -1 || m_name[m_name.Length - 1] == ' ')
```

Reply:
i.e. you just need to check the first and last chars for space

Reply:
`m_name[m_name.Length - 1] == ' '` rather than `m_name.Trim() != m_name`

Reply:
Teaches me for not building first. Resolved now.